File: app/src/main/java/com/cradletrial/cradlevhtapp/view/ui/intro/PrivacyPolicyFragment.java
package com.cradletrial.cradlevhtapp.view.ui.intro;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import com.cradletrial.cradlevhtapp.R;
import com.cradletrial.cradlevhtapp.view.ui.reading.PatientInfoFragment;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

/**
 * Confirm privacy policy with user
 */
public class PrivacyPolicyFragment extends IntroBaseFragment {
    private View mView;
    private List<CheckBox> checkBoxes = new ArrayList<>();

    public PrivacyPolicyFragment() {
        // Required empty public constructor
        TAG = PatientInfoFragment.class.getName();
    }

    public static PrivacyPolicyFragment newInstance() {
        PrivacyPolicyFragment fragment = new PrivacyPolicyFragment();
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_intro_privacy_policy, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mView = view;

        displayTextOnScreen();
    }

    private void displayTextOnScreen() {
        // source: http://www.java2s.com/Code/Android/UI/Programmaticallyloadtextfromanassetandplaceitintothetextview.htm
        // Programmatically load text from an asset and place it into the
        // text view.  Note that the text we are loading is ASCII, so we
        // need to convert it to UTF-16.
        try {
            InputStream is = getActivity().getAssets().open("CradleVSASupportAppPrivacyPolicy.txt");

            // We guarantee that the available method returns the total
            // size of the asset...  of course, this does mean that a single
            // asset can't be more than 2 gigs.
            int size = is.available();

            // Read the entire asset into a local byte buffer.
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();

            // Convert the buffer into a string.
            String text = new String(buffer);

            // Finally stick the string into the text view.
            TextView tv = getView().findViewById(R.id.tvPrivacyPolicy);
            tv.setText(text);
        } catch (IOException e) {
            // Should never happen!
            throw new RuntimeException(e);
        }
    }

    @Override
    public void onMyBeingDisplayed() {
        // may not have created view yet.
        if (mView == null) {
            return;
        }
        hideKeyboard();

    }


    @Override
    public boolean onMyBeingHidden() {
        // may not have created view yet.
        if (mView == null) {
            return true;
        }

        return true;
    }


}

File: app/src/main/java/com/cradle/neptune/view/ui/intro/PrivacyPolicyFragment.kt
package com.cradle.neptune.view.ui.intro

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.cradle.neptune.R
import com.cradle.neptune.view.ui.reading.PatientInfoFragment
import java.io.IOException

/**
 * Confirm privacy policy with user
 */
class PrivacyPolicyFragment : IntroBaseFragment() {
    private var mView: View? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_intro_privacy_policy, container, false)
    }

    override fun onViewCreated(
        view: View,
        savedInstanceState: Bundle?
    ) {
        super.onViewCreated(view, savedInstanceState)
        mView = view
        displayTextOnScreen()
    }

    private fun displayTextOnScreen() {
        // source: http://www.java2s.com/Code/Android/UI/Programmaticallyloadtextfromanassetandplaceitintothetextview.htm
        // Programmatically load text from an asset and place it into the
        // text view.  Note that the text we are loading is ASCII, so we
        // need to convert it to UTF-16.
        try {
            val inputStream =
                requireActivity().assets.open("CradleVSASupportAppPrivacyPolicy.txt")

            // We guarantee that the available method returns the total
            // size of the asset...  of course, this does mean that a single
            // asset can't be more than 2 gigs.
            val size = inputStream.available()

            // Read the entire asset into a local byte buffer.
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()

            // Convert the buffer into a string.
            val text = String(buffer)

            // Finally stick the string into the text view.
            val tv = requireView().findViewById<TextView>(R.id.tvPrivacyPolicy)
            tv.text = text
        } catch (e: IOException) {
            // Should never happen!
            throw RuntimeException(e)
        }
    }

    override fun onMyBeingDisplayed() {
        // may not have created view yet.
        if (mView == null) {
            return
        }
        hideKeyboard()
    }

    override fun onMyBeingHidden(): Boolean {
        // may not have created view yet.
        return if (mView == null) {
            true
        } else true
    }

    companion object {
        fun newInstance(): PrivacyPolicyFragment {
            return PrivacyPolicyFragment()
        }
    }

    init {
        // Required empty public constructor
        TAG = PatientInfoFragment::class.java.name
    }
}
